### PR TITLE
[DOC] Enhanced RDoc for File

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- ::pipe?
- ::symlink?
- ::socket?
- ::blockdev?
- ::chardev?

### CODE REVIEW

File: file.c [hunk 1]
```diff
@@ -1621,11 +1621,15 @@ rb_file_directory_p(VALUE obj, VALUE fname)
 
 /*
  * call-seq:
- *   File.pipe?(file_name)   ->  true or false
+ *   File.pipe?(object) -> true or false
  *
- * Returns <code>true</code> if the named file is a pipe.
+ * Returns +true+ if +object+ is a pipe, +false+ otherwise:
```
Comment (jeremyevans) [inline file.c:1626]: The main use of `File.pipe?` is with file names, though it does also support IO objects.  There should be a description and examples for use with file names (pipes in the file system can be created with `File.mkfifo`).

Similar issue for `File.socket?`

File: file.c [hunk 2]
```diff
@@ -1621,11 +1621,15 @@ rb_file_directory_p(VALUE obj, VALUE fname)
 
 /*
  * call-seq:
- *   File.pipe?(file_name)   ->  true or false
```
Comment (jeremyevans) [inline file.c:1624]: For all of these methods, they are primarily called with filenames, so the argument in the call-seq should remain `file_name`.

**Review by jeremyevans — CHANGES_REQUESTED** [review]

**Comment by BurdetteLamar** [comment]
@jeremyevans, there's a lot in File that I don't understand very well.  I'd like to close this and find something else to work on.  (Any suggestions?)

**Comment by jeremyevans** [comment]
> @jeremyevans, there's a lot in File that I don't understand very well. I'd like to close this and find something else to work on. (Any suggestions?)

Closing this is fine. I don't have any suggestions currently.

**Comment by BurdetteLamar** [comment]
Re-opening this, because it's time I learned more about class File and its methods.

**Comment by BurdetteLamar** [comment]
I thought I had re-opened this already (and the thingy says so above).

**Comment by BurdetteLamar** [comment]
Rolled forward to https://github.com/ruby/ruby/pull/5888.

**Pull request closed without merging**